airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -0,0 +1,1074 @@
+      entity.domain = domain.id;
+      entity.index = updateAirEntityCounter(AIR_DOMAIN_NEW_RESOLVER_ENTITY_COUNTER_ID, block);
+      entity.save();
+    }
+    return entity as AirDomainNewResolverTransaction;
+  }
+
+  /**
+   * @dev this function gets or creates a new air domain entity
+   * @param domain Domain class object
+   * @returns AirDomain entity
+   */
+  function getOrCreateAirDomain(
+    domain: Domain,
+  ): AirDomain {
+    let entity = AirDomain.load(domain.id);
+    if (entity == null) {
+      entity = new AirDomain(domain.id);
+      entity.subdomainCount = BIG_INT_ZERO;
+      entity.owner = getOrCreateAirAccount(domain.chainId, ZERO_ADDRESS).id;
+      entity.tokenAddress = getOrCreateAirToken(domain.chainId, domain.tokenAddress).id;
+      entity.isPrimary = false;
+      entity.isMigrated = false;
+      entity.expiryTimestamp = BIG_INT_ZERO;
+      entity.registrationCost = BIG_INT_ZERO;
+      entity.createdAt = domain.block.id;
+      entity.lastBlock = domain.block.id;
+    }
+    if (entity.id == ROOT_NODE) {
+      entity.isMigrated = true;
+    }
+    entity.save();
+    return entity as AirDomain;
+  }
+
+  /**
+   * @dev this function gets or creates a new air domain owner changed transaction entity
+   * @param block air block entity
+   * @param logIndex log index
+   * @param chainId chain id
+   * @param previousOwnerId previous owner id
+   * @param newOwner new owner address
+   * @param transactionHash transaction hash
+   * @param tokenId token id
+   * @param domain air domain
+   * @returns AirDomainOwnerChangedTransaction entity
+   */
+  function getOrCreateAirDomainOwnerChangedTransaction(
+    block: AirBlock,
+    logIndex: BigInt,
+    chainId: string,
+    previousOwnerId: string,
+    newOwner: string,
+    transactionHash: Bytes,
+    tokenId: string,
+    domain: AirDomain,
+  ): AirDomainOwnerChangedTransaction {
+    let id = createEntityId(transactionHash, block.number, logIndex);
+    let entity = AirDomainOwnerChangedTransaction.load(id);
+    if (entity == null) {
+      entity = new AirDomainOwnerChangedTransaction(id);
+      entity.previousOwner = previousOwnerId;
+      entity.newOwner = getOrCreateAirAccount(chainId, newOwner).id;
+      entity.transactionHash = transactionHash.toHex();
+      entity.tokenId = tokenId;
+      entity.domain = domain.id;
+      entity.block = block.id;
+      entity.index = updateAirEntityCounter(AIR_DOMAIN_OWNER_CHANGED_ENTITY_COUNTER_ID, block);
+      entity.save();
+    }
+    return entity as AirDomainOwnerChangedTransaction;
+  }
+
+  /**
+   * @dev this function does a recursive domain deletion for a particular domain
+   * @param domain air domain entity
+   * @param chainId chain id
+   * @param block air block entity
+   * @param tokenAddress contract address of nft token
+   * @returns parent domain id
+   */
+  function recurseDomainDelete(domain: AirDomain, chainId: string, block: AirBlock, tokenAddress: string): string | null {
```

Comment:
You are updating the subdomainCount, can we rename the function name? `recurseDomainDelete` sound like the entities are being deleted.

Reply:
done